File: 791/CustomSortString.cpp
/*
https://leetcode.com/problems/custom-sort-string/description/?envType=daily-question&envId=2024-03-11
*/

class Solution {
public:
    string customSortString(string order, string s) {
        int szs=s.size();
        set<int>seenidxs;
        string res="";
        for(auto &ordrch: order){
            for(int i=0; i<szs; ++i){
                if(seenidxs.find(i)==seenidxs.end()){
                    if(s[i]==ordrch){
                        res+=ordrch;
                        seenidxs.insert(i);
                    }
                }
            }
        }
        for(int i=0; i<szs; ++i){
            if(seenidxs.find(i)==seenidxs.end()){
                res+=s[i];
            }
        }
        return res;
    }
};

File: 7/791/CustomSortString.ts
/*
https://leetcode.com/problems/custom-sort-string/description/?envType=daily-question&envId=2024-03-11
*/

function customSortString(order: string, s: string): string {
    let szordr:number=order.length, szs:number=s.length;
    let seenidxs:Set<number>=new Set();
    let res:string="";
    for(let i:number=0; i<szordr; ++i){
        for(let j:number=0; j<szs; ++j){
            if(!seenidxs.has(j)){
                if(s[j]===order[i]){
                    res+=order[i];
                    seenidxs.add(j);
                }
            }
        }
    }
    for(let i:number=0; i<szs; ++i){
        if(!seenidxs.has(i)){
            res+=s[i];
        }
    }
    return res;
};